src/components/Radio/index.test.tsx

```diff
@@ -0,0 +1,283 @@
+import React, { createRef, useState } from 'react'
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { Radio, RadioGroup, RadioWithLabel } from './'
+
+const RadioExample = () => {
+  const [value, setValue] = useState(null)
+
+  return (
+    <RadioGroup
+      options={[
+        { id: 'Rob', value: 'Rob', label: 'Rob' },
+        { id: 'Bob', value: 'Bob', label: 'Bob' },
+      ]}
+      onChange={e => setValue(e.target.value)}
+      value={value}
+      legend="This is a RadioGroup"
+      name="who-is-best"
+    />
+  )
+}
+
+describe('Radio', () => {
+  it('renders without crashing', () => {
+    render(
+      <Radio onChange={() => null} value="on" checked id="test" name="test" />
+    )
+
+    expect(
+      screen.getByRole('presentation', {
+        name: /checkmark/i,
+      })
+    ).toBeInTheDocument()
+  })
+
+  describe('when it is clicked', () => {
+    describe('and it has not been checked before', () => {
+      it('fires an on change event ', () => {
+        const handleChange = jest.fn(event => event.target.value)
+        const valueMock = 'test'
+        const ref = createRef<HTMLInputElement>()
+
+        render(
+          <Radio
+            ref={ref}
+            onChange={handleChange}
+            value={valueMock}
+            checked={false}
+            id="test"
+            name="test"
+          />
+        )
+
+        userEvent.click(screen.getByRole('radio'))
+
+        expect(handleChange).toBeCalledTimes(1)
+        expect(handleChange).toHaveBeenCalledWith(
+          expect.objectContaining({
+            target: expect.objectContaining({
+              value: valueMock,
+            }),
+          })
+        )
+      })
+    })
+
+    describe('and it is already checked', () => {
+      it('does not fire an on change event', () => {
+        const handleChange = jest.fn(event => event.target.value)
+        const ref = createRef<HTMLInputElement>()
+
+        render(
+          <Radio
+            ref={ref}
+            onChange={handleChange}
+            value="test"
+            checked={true}
+            id="test"
+            name="test"
+          />
+        )
+
+        userEvent.click(ref.current)
```

Comment:
**suggestion:** What do you think about clicking on the label instead? (getByRole ...) It would resemble the user experience more closely.